src/components/viewStats/components/ViewStats.js

```diff
@@ -18,56 +21,122 @@ const countGender = (oldCount, person) => {
     return newCount;
 };
 
+// reducer function used to group the candidiates by age
+const countAge = (oldCount, person) => {
+    const newCount = oldCount;
+    newCount['18-25'] += (person.age >= 18 && person.age < 26);
+    newCount['26-40'] += (person.age >= 26 && person.age < 40);
+    newCount['40-60'] += (person.age >= 40 && person.age < 60);
+    newCount['60+'] += (person.age > 60);
+    return newCount;
+};
+// function to transfor candidates from their id to their name
+const transformCandidates = candidates => {
+    const nameMap = {};
+    candidates.forEach(candidate => {
```

Comment:
use a `.reduce` instead of forEach here.
```
const transformCandidates = candidates => candidates.reduce((total, candidate, => {
    ....
}, {});
```